fmriprep/interfaces/fmap.py

```diff
@@ -497,6 +539,56 @@ def phdiff2fmap(in_file, delta_te, newpath=None):
     return out_file
 
 
+def phases2fmap(phase_files, magnitude_files, metadatas, newpath=None):
+    """calculates a phasediff from two phase images. Assumes monopolar
+    readout and that the second image in the list corresponds to the
+    longer echo time"""
+    import numpy as np
+    import nibabel as nb
+    from nipype.utils.filemanip import fname_presuffix
+    from copy import deepcopy
+
+    phasediff_file = fname_presuffix(phase_files[0], suffix='_phasediff',
+                                     newpath=newpath)
+    echo_times = [meta.get("EchoTime") for meta in metadatas]
+    if None in echo_times or echo_times[0] == echo_times[1]:
+        raise RuntimeError()
+    # Determine the order of subtraction
+    short_echo_index = echo_times.index(min(echo_times))
+    long_echo_index = echo_times.index(max(echo_times))
+
+    magnitude_image = magnitude_files[short_echo_index]
+    short_phase_image = phase_files[short_echo_index]
+    long_phase_image = phase_files[long_echo_index]
+
+    mag_img = nb.load(magnitude_image)
+    mag = mag_img.get_data().astype(np.float)
+    image0 = nb.load(short_phase_image)
+    phase0 = image0.get_data().astype(np.float)
+    image1 = nb.load(long_phase_image)
+    phase1 = image1.get_data().astype(np.float)
```

Comment:
You can use

```Python
phase1 = image1.get_fdata()
```

Or to be more concise, you can just replace `phase1` later with `image1.get_fdata()`.

Same comments apply to `phase0` and `mag` (`get_fdata()` is caching, so calling multiple times is cheap), though I also have a question about whether `mag` is needed at all.